transaction/src/manifest/parser.rs

```diff
@@ -439,20 +493,27 @@ impl Parser {
     }
 
     pub fn peek(&mut self) -> Result<Token, ParserError> {
-        self.tokens
-            .get(self.current)
-            .cloned()
-            .ok_or(ParserError::UnexpectedEof)
+        self.tokens.get(self.current).cloned().ok_or(ParserError {
+            error_kind: ParserErrorKind::UnexpectedEof,
+            span: match self.tokens.get(self.previous) {
+                Some(token) => Span {
+                    start: token.span.end,
+                    end: token.span.end,
+                },
+                None => span!(start = (0, 0, 0), end = (0, 0, 0)),
```

Comment:
> I think this should be unreachable unless tokens has 0 length - but perhaps this should be validated in hte constructor?

Added tokens length validation in the constructor and using `self.tokens[self.current - 1].span.end` here.

> A span with line_number = 0 isn't valid. Perhaps this might give weight to use changing line number => line index. Or potentially implementing Default on span to start with line_number = 1.

Refactored line number to line index anyway.